AppCenterAnalytics/AppCenterAnalytics/Internals/PropertyConfigurator/MSPropertyConfigurator.m

```diff
@@ -0,0 +1,64 @@
+#import "MSAnalyticsTransmissionTargetPrivate.h"
+#import "MSCommonSchemaLog.h"
+#import "MSPropertyConfigurator.h"
+#import "MSPropertyConfiguratorPrivate.h"
+
+@implementation MSPropertyConfigurator
+
+- (instancetype)initWithTransmissionTarget:(MSAnalyticsTransmissionTarget *)transmissionTarget {
+  if ((self = [super init])) {
+    _transmissionTarget = transmissionTarget;
+  }
+  return self;
+}
+
+- (void)setAppVersion:(NSString *)appVersion {
+  _appVersion = appVersion;
+}
+
+- (void)setAppName:(NSString *)appName {
+  _appName = appName;
+}
+
+- (void)setAppLocale:(NSString *)appLocale {
+  _appLocale = appLocale;
+}
+
+#pragma mark - MSChannelDelegate
+
+- (void)channel:(id<MSChannelProtocol>)__unused channel prepareLog:(id<MSLog>)log {
+  if ([log isKindOfClass:[MSCommonSchemaLog class]] && [self.transmissionTarget isEnabled]) {
+
+    // Override the application version.
```

Comment:
I was thinking that there is probably a way to share code using [KVC](https://developer.apple.com/library/archive/documentation/Cocoa/Conceptual/KeyValueCoding/index.html). Since it's all about updating `propertyConfigurator` properties. But I'll let you decide. Not a blocker.

Reply:
Same discussion happened for Android implementation. For now we only have three so it is kinda okay but we should definitely consider better option when we add more properties here. Won't fix just for now.

Reply:
@clpolet good point.  We don't like that many while loops either.  We can add a TODO for now until we come up with something better.